provisioning/baremetal_secrets_test.go

```diff
@@ -0,0 +1,87 @@
+package provisioning
+
+import (
+	"testing"
+
+	"golang.org/x/net/context"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakekube "k8s.io/client-go/kubernetes/fake"
+)
+
+const testNamespace = "test-namespce"
+
+func TestGenerateRandomPassword(t *testing.T) {
+	pwd, err := generateRandomPassword()
+	if err != nil {
+		t.Errorf("Unexpected error: %s", err)
+	}
+	if pwd == "" {
+		t.Errorf("Expected a valid string but got null")
+	}
```

Comment:
I think that an interesting test for a random password generator method would be to verify that two consecutive calls produce different passwords

Reply:
Added test case to try 2 consecutive calls. Ideally, the code to generate random numbers of a reasonable length should not repeat for n tries where n is a non-trivial number.